File: lemur-app/src/main/java/io/github/malczuuu/lemur/app/domain/player/PlayerEvent.java
package io.github.malczuuu.lemur.app.domain.player;

import io.github.malczuuu.lemur.app.domain.Event;

public interface PlayerEvent extends Event {

  String playerId();

  record PlayerBanned(String playerId) implements PlayerEvent {}

  record PlayerCreated(String playerId) implements PlayerEvent {}

  record PlayerRatingChanged(String playerId, int oldRating, int newRating)
      implements PlayerEvent {}

  record PlayerUnbanned(String playerId) implements PlayerEvent {}

  record PlayerUpdated(String playerId) implements PlayerEvent {}
}

File: lemur-app/src/main/kotlin/io/github/malczuuu/lemur/app/domain/player/PlayerEvent.kt
package io.github.malczuuu.lemur.app.domain.player

import io.github.malczuuu.lemur.app.domain.Event

interface PlayerEvent : Event {

    data class PlayerCreated(val playerId: String) : Event

    data class PlayerUpdated(val playerId: String) : Event

    data class PlayerRatingChanged(val playerId: String, val oldRating: Int, val newRating: Int) : Event

    data class PlayerBanned(val playerId: String) : Event

    data class PlayerUnbanned(val playerId: String) : Event
}
